packages/adapters-library/src/defiProvider.ts

```diff
@@ -61,14 +61,23 @@ function buildMetadataProviders(): Record<Chain, IMetadataProvider> {
 const dbParams = (chainId: Chain): [string, Database.Options] => {
   const dbPath = path.join(__dirname, '../../..', `${ChainName[chainId]}.db`)
 
-  if (!existsSync(dbPath)) {
+  if (
+    !(process.env.DEFI_ALLOW_DB_CREATION !== 'false') &&
+    !existsSync(dbPath)
+  ) {
```

Comment:
In what event returning a `dbPath` that doesn't exist will work?

Reply:
If database creation is not allowed and the database file doesn't exist, it'll throw